notes.md

```diff
@@ -0,0 +1,215 @@
+# Why does `:` only sometimes work?
+I don't like `:` being listed as a `source_column_selector` when it often errors.
+
+`select(df, :)`
+: works
+
+`transform(df, Cols(:) => ByRow(max))`
+: works
+
+`transform(df, : => ByRow(max))` errors
+: ERROR: syntax: whitespace not allowed after ":" used for quoting
+
+Similarly, regular expressions also sometimes must be wrapped in `Cols`.
+This seems inconsistent.
+I don't understand the rules for when `Cols` is and isn't needed.
```

Comment:
The only two syntax problems are the ones you asked above. This can be added in your PR.